src/XamarinUsbDriver/UsbSerial/ProlificSerialPort.cs

```diff
@@ -115,26 +175,34 @@ public override void Open(UsbDeviceConnection connection)
                     }
                 }
 
-                if (Device.DeviceClass == (UsbClass) 0x02)
-                {
-                    mDeviceType = DEVICE_TYPE_0;
-                }
-                else
+                SetControlLines(mControlLinesValue);
+
+                if (type != PL2303Type.TYPE_HXN)
                 {
-                    if ((Device.DeviceClass == 0x00)
-                        || (Device.DeviceClass == (UsbClass) 0xff))
+                    VendorIn(0x8484, 0, 1);
+
+                    VendorOut(0x0404, 0, null);
+
+                    VendorIn(0x8484, 0, 1);
+                    VendorIn(0x8383, 0, 1);
+                    VendorIn(0x8484, 0, 1);
+
+                    VendorOut(0x0404, 1, null);
+
+                    VendorIn(0x8484, 0, 1);
+                    VendorIn(0x8383, 0, 1);
```

Comment:
This is sort of a bug fix. The old `DoBlackMagic()` method just read `0x8484` twice, but the Linux driver reads `0x8484` then `0x8383`.